### PR TITLE
Dala 4836 errors in community manager

### PR DESCRIPTION
**Pull request by stewal19**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt [hunk 1]
```diff
@@ -367,4 +367,27 @@ interface CompanyApi {
...
     fun getCompanyInfo(@PathVariable("companyId") companyId: String): ResponseEntity<CompanyInformation>
+
+    /**
+     * A method to check if a companyId is valid
+     * @param companyId the identifier
+     */
+    @Operation(
+        summary = "Checks that a companyId exists.",
+        description = "Checks that a companyId exists.",
+    )
```
Comment (LaurinStrelow) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:378]: Bitte umbennen in 'Checks if a ...'

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt [hunk 2]
```diff
@@ -367,4 +367,27 @@ interface CompanyApi {
...
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully checked that companyId exists."),
+            ApiResponse(responseCode = "404", description = "Successfully checked that companyIdr does not exist."),
+        ],
```
Comment (LaurinStrelow) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:383]: companyId statt companyIdr. Ich find die description auch wiedersprüchlich. Vllt.: Check was successful, but companyId does not exist

File: dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/CompanyIdValidator.kt [hunk 3]
```diff
@@ -34,4 +34,24 @@ class CompanyIdValidator(
             } else { throw e }
         }
     }
+
+    /**
+     * Checks if a companyId exists on Dataland by trying to retrieve it in the backend.
+     * If it does not exist the method catches the not-found-exception from the backend and throws a
+     * resource-not-found exception here in the community manager.
+     * @param companyId is the companyId to check for
+     * @returns the name of the company if it could be found
```
Comment (LaurinStrelow) [inline dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/CompanyIdValidator.kt:43]: Die Funktion sollte nichts zurückgeben, oder?

Comment (stewal19) [reply dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/CompanyIdValidator.kt:43]: Geändert

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/CompanyDataController.kt [hunk 4]
```diff
@@ -203,4 +203,7 @@ class CompanyDataController(
                 .getCompanyApiModelById(companyId, DatalandAuthentication.fromContextOrNull()).companyInformation,
         )
     }
+    override fun isCompanyIdValid(companyId: String) {
+        companyQueryManager.verifyCompanyIdExists(companyId)
+    }
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/CompanyDataController.kt:208]: Die gab es schon im Backend

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt (hunk 1 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:378]: Geändert

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt (hunk 2 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:383]: Typo changed. 
Bei dem Check ist es so, der ist halt auch successfull wenn er herausfindet, dass es die Firma nicht gibt. Ein nicht successfull request wäre, wenn der mit einem 500 Fehler fehlschlägt.

Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:391]: Das versteh ich. Mein Vorschlag hatte sich auf den zweiten Fall bezogen. Wir können es aber auch so lassen.

Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:391]: Dann ändere ich das zu: Check was successfull. Company is not known by dataland.
Passt das für dich?

Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt:390]: Ja

**Pull request merged**